torch_em/data/datasets/light_microscopy/cellpose.py

```diff
@@ -0,0 +1,154 @@
+"""This dataset contains annotation for cell segmentation in
+fluorescene microscently-labeled microscopy images.
+
+This dataset is from the following publications:
+- https://doi.org/10.1038/s41592-020-01018-x
+- https://doi.org/10.1038/s41592-022-01663-4
+Please cite it if you use this dataset in your research.
+"""
+
+import os
+from glob import glob
+from natsort import natsorted
+from typing import Union, Tuple, Literal
+
+import torch_em
+
+from torch.utils.data import Dataset, DataLoader
+
+from .. import util
+from .neurips_cell_seg import to_rgb
+
+
+URL = "https://www.cellpose.org/dataset"
+
+
+def _get_cellpose_paths(data_dir):
+    image_paths = natsorted(glob(os.path.join(data_dir, "*_img.png")))
+    gt_paths = natsorted(glob(os.path.join(data_dir, "*_masks.png")))
+
+    return image_paths, gt_paths
+
+
+def get_cellpose_data(
+    path: Union[os.PathLike, str],
+    split: Literal["train", "test"],
+    choice: Literal["cyto", "cyto2"],
+    download: bool = False,
+):
+    """Instruction to download CellPose data.
+
+    Args:
+        path: Filepath to a folder where the data is downloaded for further processing.
+        split: The data split to use. Either 'train', or 'test'.
+        choice: The choice of dataset. Either 'cyto' or 'cyto2'.
+        download: Whether to download the data if it is not present.
+
+    Returns:
+        The filepath to the data.
+    """
+    if download:
+        assert NotImplementedError(
+            "The dataset cannot be automatically downloaded. ",
+            "Please see 'get_cellpose_data' in 'torch_em/data/datasets/cellpose.py' for details."
+        )
+
+    per_choice_dir = os.path.join(path, choice)  # path where the unzipped files will be stored
+    if choice == "cyto":
+        assert split in ["train", "test"], f"'{split}' is not a valid split in '{choice}'."
+        zip_path = os.path.join(path, f"{split}.zip")
+        data_dir = os.path.join(per_choice_dir, split)  # path where the per split images for 'cyto' exist.
+    elif choice == "cyto2":
+        assert split == "train", f"'{split}' is not a valid split in '{choice}'."
+        zip_path = os.path.join(path, "train_cyto2.zip")
+        data_dir = os.path.join(per_choice_dir, "train_cyto2")  # path where 'train' split images for 'cyto2' exist.
+    else:
+        raise ValueError(f"'{choice}' is not a valid dataset choice.")
+
+    if not os.path.exists(data_dir):
+        util.unzip(zip_path=zip_path, dst=per_choice_dir, remove=False)
+
+    return data_dir
+
+
+def get_cellpose_dataset(
+    path: Union[os.PathLike, str],
+    split: Literal["train", "test"],
+    patch_shape: Tuple[int, int],
+    choice: Literal["cyto", "cyto2"] = "cyto",
```

Comment:
Wouldn't we rather use `cyto2` as the default, since it's the larger dataset?

Reply:
Looks like `cyto` has more number of images (540) compared to `cyto2` (256).

Reply:
Ah, and `cyto` has provided train-test splits (unlike for `cyto2`, where we only have the train split provided)

Reply:
I like the idea of supporting the possibility of combining both datasets (and is also possible as the image ids do not have any overlaps).

I've extended this feature in the new commit. Thanks for the hint!